EL-515: Stop trying to assess smod value prematurely

patrick-laa opened the pull request:
## What

[Link to story](https://dsdmoj.atlassian.net/browse/EL-515)

We were trying to sum the values of SMOD assets before we'd assessed the values of those assets, leading to incorrect calculations. This fixes that by moving the summing to later in the process.

starswan commented on app/services/calculators/subject_matter_of_dispute_disregard_calculator.rb:
```diff
@@ -1,30 +1,28 @@
+# WARNING: This calculator assumes that the assessed value/equity of all disputed properties and vehicles
+# has already been calculated. If this is not the case, it will produce inaccurate results.
 module Calculators
```
I think we're (still) missing a test of the top level that proves that everything is joined up properly. This comment just shows how fragile the current structure is - effectively you can't tell at level 2 whether level 1 has done the calculations you require

patrick-laa replied:
Good point!

patrick-laa commented:
As per discussion, I've modified the DB to allow 'assessed' value fields to be null if there is not a meaningful value to put in them, tried to make CapitalCollator be explicit about when it's making changes to the database that other bits of it rely on, and added a simple Cucumber feature file to prove everything works.
